test/powershell/Modules/Microsoft.PowerShell.Diagnostics/Get-Counter.Tests.ps1

```diff
@@ -0,0 +1,231 @@
+            }
+        )
+
+        foreach ($testCase in $parameterTestCases)
+        {
+            ValidateParameters($testCase)
+        }
+    }
+
+    Context "Get-Counter CounterSet tests" {
+        It "Get-Counter with no parameters returns data for a default set of counters" {
+            $counterData = Get-Counter
+            # At the very least we should get processor and memory
+            $counterData.CounterSamples.Length | should BeGreaterThan 1
+        }
+
+        It "Can retrieve the specified number of counter samples" {
+            $counterPath = TranslateCounterPath("\Memory\Available Bytes")
+            $counterCount = 5
+            $counterData = Get-Counter -Counter $counterPath -MaxSamples $counterCount
+            $counterData.Length | Should Be $counterCount
+        }
+
+        It "Can specify the sample interval" {
+            $counterPath = TranslateCounterPath("\PhysicalDisk(*)\Current Disk Queue Length")
+            $counterCount = 5
+            $sampleInterval = 2
+            $startTime = Get-Date
+            $counterData = Get-Counter -Counter $counterPath -SampleInterval $sampleInterval -MaxSamples $counterCount
+            $endTime = Get-Date
+            $counterData.Length | Should Be $counterCount
+            ($endTime - $startTime).TotalSeconds | Should Not BeLessThan ($counterCount * $sampleInterval)
+        }
+
+        It "Can process array of counter names" {
+            $counterPaths = @((TranslateCounterPath("\PhysicalDisk(_Total)\Disk Read Bytes/sec")),
+                              (TranslateCounterPath("\Memory\Available bytes")))
+            $counterData = Get-Counter -Counter $counterPaths
+            $counterData.CounterSamples.Length | Should Be $counterPaths.Length
+        }
+    }
+
+    Context "Get-Counter ListSet tests" {
+        It "Can retrieve specified counter set" {
+            $counterSetName = "Memory"
+            $counterSet = Get-Counter -ListSet $counterSetName
+            $counterSet.Length | Should Be 1
+            $counterSet.CounterSetName | Should Be $counterSetName
+        }
+
+        It "Can process an array of counter set names" {
+            $counterSetNames = @("Memory", "Processor")
+            $counterSets = Get-Counter -ListSet $counterSetNames
+            $counterSets.Length | Should Be 2
+            $counterSets[0].CounterSetName | Should Be $counterSetNames[0]
+            $counterSets[1].CounterSetName | Should Be $counterSetNames[1]
+        }
+
+        # This test should work for English, but other languages are
+        # problematic since there is no reasonable way to construct
+        # a wild-card pattern that will, for every language, result
+        # in a known set of values or evan a set with a known minimum
+        # number of items.
+        It "Can process counter set name with wildcards" {
+            $wildcardBase = "roc"
```

Comment:
These test should then check to make sure that the current language is English and then conditionally execute if it matches. -Skip works well for this technique.